src/libraries/System.Private.CoreLib/gen/IntrinsicsInSystemPrivateCoreLibAnalyzer.cs

```diff
@@ -0,0 +1,693 @@
+                }
+            }
+        }
+
+        private static bool ConditionAllowsSymbol(ISymbol symbolOfInvokeTarget, INamedTypeSymbol namedTypeThatIsSafeToUse, IntrinsicsAnalyzerOnLoadData onLoadData)
+        {
+            HashSet<INamedTypeSymbol> examinedSymbols = new HashSet<INamedTypeSymbol>(SymbolEqualityComparer.Default);
+            Stack<INamedTypeSymbol> symbolsToExamine = new Stack<INamedTypeSymbol>();
+            symbolsToExamine.Push(namedTypeThatIsSafeToUse);
+
+            while (symbolsToExamine.Count > 0)
+            {
+                INamedTypeSymbol symbol = symbolsToExamine.Pop();
+                if (symbolOfInvokeTarget.ContainingSymbol.Equals(symbol, SymbolEqualityComparer.Default))
+                    return true;
+
+                foreach (var helperForType in GetCompExactlyDependsOnUseList(symbolOfInvokeTarget, onLoadData))
+                {
+                    if (helperForType.Equals(symbol, SymbolEqualityComparer.Default))
+                        return true;
+                }
+
+                examinedSymbols.Add(symbol);
+                if (symbol.ContainingType != null && !examinedSymbols.Contains(symbol.ContainingType))
+                    symbolsToExamine.Push(symbol.ContainingType);
+                if (symbol.BaseType != null && !examinedSymbols.Contains(symbol.BaseType))
+                    symbolsToExamine.Push(symbol.BaseType);
+            }
+
+            return false;
+        }
+
+        private static bool TypeSymbolAllowsTypeSymbol(INamedTypeSymbol namedTypeToCheckForSupport, INamedTypeSymbol namedTypeThatProvidesSupport)
+        {
+            HashSet<INamedTypeSymbol> examinedSymbols = new HashSet<INamedTypeSymbol>(SymbolEqualityComparer.Default);
+            Stack<INamedTypeSymbol> symbolsToExamine = new Stack<INamedTypeSymbol>();
+            symbolsToExamine.Push(namedTypeThatProvidesSupport);
+
+            while (symbolsToExamine.Count > 0)
+            {
+                INamedTypeSymbol symbol = symbolsToExamine.Pop();
+                if (namedTypeToCheckForSupport.Equals(symbol, SymbolEqualityComparer.Default))
+                    return true;
+
+                examinedSymbols.Add(symbol);
+                if (symbol.ContainingType != null && !examinedSymbols.Contains(symbol.ContainingType))
+                    symbolsToExamine.Push(symbol.ContainingType);
+                if (symbol.BaseType != null && !examinedSymbols.Contains(symbol.BaseType))
+                    symbolsToExamine.Push(symbol.BaseType);
+            }
+
+            return false;
+        }
+
+        private static INamespaceSymbol? SymbolToNamespaceSymbol(ISymbol symbol)
+        {
+            if (symbol.ContainingType != null)
+                return SymbolToNamespaceSymbol(symbol.ContainingType);
+
+            if (symbol.ContainingNamespace != null)
+            {
+                return symbol.ContainingNamespace;
+            }
+
+            return null;
+        }
```

Comment:
Is there a reason we can't just use `ContainingNamespace` here? It should work for all symbols, not just type symbols IIRC.

```suggestion
        {
            return symbol.ContainingNamespace;
        }
```